pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/Loader.java

```diff
@@ -48,7 +49,11 @@ static class UrlLoader extends Loader {
         @Override
         @Nullable
         InputStream getInputStream() throws IOException {
-            return url.openStream();
+            URLConnection connection = url.openConnection();
+            // disable the cache, so that the underlying JarFile (if any) is closed, when
+            // the returned stream is closed - leaving no open files behind.
+            connection.setUseCaches(false);
```

Comment:
Note: This is obsolete with the changes in #4628 - if the resources are loaded with URLClassLoader#getResourceAsStream, then the URLClassLoader will take care of closing the streams and the underlying jar file, when the class loader is closed. Only when dealing with URLs directly we need to be careful here.